src/App.tsx

```diff
@@ -1,26 +1,37 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React, { useReducer } from 'react'
+import SearchPage from './components/SearchPage'
+import MyCollectionsPage from './components/MyCollectionsPage'
+import Route from './components/Route'
+import Link from './components/Link'
+import { initialCollection, collectionReducer } from 'utils'
 
 function App() {
+  const [myCollection, dispatch] = useReducer(
```

Comment:
This is a really good use case for using Context (you're prop-drilling this hardcore). Try putting this in Context!